### DISCUSSION TITLE
command --watch not working as expected

### DISCUSSION THREAD

**Discussion by troyyer** [webdriverio/webdriverio#12024] [Q&A]
I am having an issue trying to use the '--watch' feature (https://webdriver.io/docs/watcher)
i saw this bug item https://github.com/webdriverio/webdriverio/issues/9612 but was unsure if it has been resolved as yet

Currently when i run a website test script using the Test Runner and with the watch command enabled, once the test has completed, the browser window will stay open (as expected), then when I try to update the test script file and after saving the file i get an error in the Terminal

```
2024-01-15T23:01:14.917Z INFO @wdio/local-runner: Start worker 0-0 with arg: run,./wdio.conf.js,--watch
[0-0] 2024-01-15T23:01:16.512Z INFO @wdio/local-runner: Run worker command: run
[0-0] 2024-01-15T23:01:17.665Z ERROR @wdio/runner: Error: You are trying to attach to a protocol stub, this should never occur, please file an issue.
[0-0]     at ProtocolStub.attachToSession (file:///C:/code/sample/node_modules/webdriverio/build/protocol-stub.js:31:19)
[0-0]     at attach (file:///C:/code/sample/node_modules/webdriverio/build/index.js:70:27)        
[0-0]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[0-0] RUNNING in chrome - file:///C:/code/sample/test/specs/test.e2e.js
[0-0] 2024-01-15T23:01:17.985Z INFO webdriver: COMMAND navigateTo("https://the-internet.herokuapp.com/dynamic_loading")
[0-0] 2024-01-15T23:01:17.986Z INFO webdriver: [POST] http://0.0.0.0:52056/session/f8d10edf46f9c1a82a570da029789b4a/url
[0-0] 2024-01-15T23:01:17.986Z INFO webdriver: DATA { url: 'https://the-internet.herokuapp.com/dynamic_loading' }
[0-0] FAILED in chrome - file:///C:/code/sample/test/specs/test.e2e.js
2024-01-15T23:01:18.164Z INFO @wdio/cli:launcher: Run onWorkerEnd hook

```

**Comment by christian-bromann** [comment]
Can you provide a minimal reproducible example?

**Comment by troyyer** [comment]
the reporducable sample is a 2 step process actually.
for browser based testing my repo will do which i updated the package json to have an additional script
https://github.com/troyyer/wdiosampleforwaituntil
after installation
```
npm run watch
```
once the first test passed
edit the test script in '/test/specs/test.e2e.js' by making line 13 to read

`  it.only("tests dynamic loading", async () => {`
after saving, watch will rerun the file and the terminal will show the error

I also ran it for component testing 
i ran the cli installer with component testing
```
 npm init wdio . 
```
with the configurations below
```
===============================
🤖 WDIO Configuration Wizard 🧙
===============================

? What type of testing would you like to do? Component or Unit Testing - in the browser
    > https://webdriver.io/docs/component-testing
? Which framework do you use for building components? Lit (https://lit.dev/)
? Which framework do you want to use? Mocha (https://mochajs.org/)
? Do you want to use a compiler? No!
? Do you want WebdriverIO to autogenerate some test files? Yes
? What should be the location of your spec files? C:\code\wdio-component\src\**\*.test.js
? Which reporter do you want to use? spec
? Do you want to add a plugin to your test setup?
? Do you want to add a service to your test setup?
```
dependencies picked up by the cli installation
```
    "@wdio/browser-runner": "^8.27.2",
    "@wdio/cli": "^8.27.2",
    "@wdio/mocha-framework": "^8.27.2",
    "@wdio/spec-reporter": "^8.27.2",
    "lit": "^3.1.1"
```

after install run wdio with watch on 
```
npm run wdio -- -- --watch
```

after the initial run has completed, go into file "{ProjectRoot}\src\Component.test.js"
edit line 20 to now read 
`        await expect(button).not.toHaveText('count is 2') `
 after saving the file i get an error in the Terminal
![rerun with wdio watch flag](https://github.com/webdriverio/webdriverio/assets/28956675/8779a7b4-6c0f-4b79-b63b-e0bd2eb422a8)

**Reply by christian-bromann** [reply]
Thank you, I will take a look into it.

**Reply by troyyer** [reply]
Thanks @christian-bromann also tangent question but is there any ticket/future plan to extend/incorporate the component testing browser so that we can use that with E2E testing? would seem like a terrific feature to have for debugging especially with the inbuilt console allowing for webdriverio commands

**Reply by christian-bromann** [reply]
There is no plan to inject similar capabilities in an e2e session, however we are working on debug tools and hope to be able to share something soon.

**Reply by christian-bromann** [reply]
@troyyer I was able to verify the issue. It seems that all works fine, even with the error.

Would you mind create an issue for this, using the detailed description you provided?

Thank you!
